src/supervised/distance.jl

```diff
@@ -108,32 +108,152 @@ islipschitzcont_deriv(::L2DistLoss) = true
 isconvex(::L2DistLoss) = true
 isstronglyconvex(::L2DistLoss) = true
 
+
+# ===========================================================
+# L(y, t) = 1 - cos((y-t)*2π/c)
+
+"""
+`PeriodicLoss <: DistanceLoss`
+
+Measures distance on a circle of specified circumference.
+"""
+immutable PeriodicLoss{T<:AbstractFloat} <: DistanceLoss
+    k::T   # k = 2π/circumference
+    function PeriodicLoss(circ::T)
+        circ > 0 || error("circumference should be strictly positive")
+        new(convert(T, 2π/circ))
+    end
+end
+PeriodicLoss{T<:AbstractFloat}(circ::T=1.0) = PeriodicLoss{T}(circ)
+PeriodicLoss(circ) = PeriodicLoss{Float64}(Float64(circ))
+
+value{T<:Number}(loss::PeriodicLoss, difference::T) = 1 - cos(difference*loss.k)
+deriv{T<:Number}(loss::PeriodicLoss, difference::T) = loss.k * sin(difference*loss.k)
+deriv2{T<:Number}(loss::PeriodicLoss, difference::T) = abs2(loss.k) * cos(difference*loss.k)
+function value_deriv{T<:Number}(loss::PeriodicLoss, difference::T)
+    dk = difference*loss.k
+    return 1-cos(dk), loss.k*sin(dk)
+end
+
+isdifferentiable(::PeriodicLoss) = true
+isdifferentiable(::PeriodicLoss, at) = true
+istwicedifferentiable(::PeriodicLoss) = true
+istwicedifferentiable(::PeriodicLoss, at) = true
+islipschitzcont(::PeriodicLoss) = true
+islipschitzcont_deriv(::PeriodicLoss) = true
+isconvex(::PeriodicLoss) = false
+isstronglyconvex(::PeriodicLoss) = false
+
+
+# ===========================================================
+# L(y, t) =  1/2 * (y-t)²    , if |y-t| < d
+# L(y, t) =  d⋅(|y-t| - d/2) , otherwise
+
+"""
+`HuberLoss <: DistanceLoss`
+
+Loss function commonly used for robustness to outliers
+"""
+type HuberLoss{T<:AbstractFloat} <: DistanceLoss
+    d::T   # boundary between quadratic and linear loss
+    function HuberLoss(d::T)
+        d > 0 || error("Huber crossover parameter must be strictly positive.")
+        new(d)
+    end
+end
+HuberLoss{T<:AbstractFloat}(d::T=1.0) = HuberLoss{T}(d)
+HuberLoss(d) = HuberLoss{Float64}(Float64(d))
+
+function value{T1,T2<:Number}(loss::HuberLoss{T1}, difference::T2)
+    T = promote_type(T1,T2)
+    abs_diff = abs(difference)
+    if abs_diff <= loss.d
+        return T(0.5)*abs2(difference)   # quadratic
```

Comment:
Yes, this is a stylistic difference I have with a lot of the Julia community. I'm not always consistent about it though.